File: pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/java/models/consumer/datasync/user/DataSyncGetUsersResult.java
package com.pubnub.api.java.models.consumer.datasync.user;

import com.pubnub.api.java.models.consumer.datasync.PNDataSyncPage;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;
import org.jetbrains.annotations.NotNull;

import java.util.List;

/**
 * Result of {@code pubnub.dataSync().getUsers()}.
 * DataSync list endpoints are cursor-based (unlike App Context's page-based paging).
 */
@Getter
@NoArgsConstructor
@ToString
public class DataSyncGetUsersResult {
    private int status;
    private List<DataSyncUser> data;

    /**
     * Cursor-based pagination metadata. Always present (an empty page when the server sends no
     * {@code meta}); read {@link PNDataSyncPage#isHasNext()} to detect the last page.
     */
    @NotNull
    private PNDataSyncPage next;

    public DataSyncGetUsersResult(int status, List<DataSyncUser> data, @NotNull PNDataSyncPage next) {
        this.status = status;
        this.data = data;
        this.next = next;
    }
}

File: pubnub-kotlin/pubnub-kotlin-api/src/commonMain/kotlin/com/pubnub/api/models/consumer/datasync/user/DataSyncGetUsersResult.kt
package com.pubnub.api.models.consumer.datasync.user

import com.pubnub.api.models.consumer.datasync.PNDataSyncPage

/**
 * Result of `pubnub.dataSync.getUsers`.
 *
 * DataSync list endpoints are **cursor-based** (unlike App Context's page-based `PNPage`).
 *
 * @property status HTTP status code of the response.
 * @property data The users on this page.
 * @property next Cursor-based pagination metadata. Always present (an empty page when the server sends
 *   no `meta`); read [PNDataSyncPage.hasNext] to detect the last page, never `next == null`.
 */
data class DataSyncGetUsersResult(
    val status: Int,
    val data: List<DataSyncUser>,
    val next: PNDataSyncPage,
)
